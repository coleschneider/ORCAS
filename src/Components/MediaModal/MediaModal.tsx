import * as React from 'react';
import ReactModal from 'react-modal';
// import ReactPlayer from 'react-player';
import { RouteComponentProps } from 'react-router';
import Loader from 'Common/Loader/Loader';
import ReactPlayer from 'react-player';

(() => {
  if (process.env.NODE_ENV === 'test') {
    return;
  }
  ReactModal.setAppElement('#root');
})();

const customStyles = {
  content: {
    // width: '1000px',
    // alignItems: 'stretch',
    borderRadius: '5px',
    bottom: 'auto',
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column',
    flexShrink: 1,
    height: '80vh',
    justifyContent: 'center',
    left: '50%',
    overflowX: 'hidden',
    overflowY: 'hidden',
    position: 'absolute',
    backgroundColor: 'black',
    right: 'auto',
    top: '50%',
    transform: 'translate(-50%, -50%)',
  },
  overlay: {
    backgroundColor: 'rgba(0, 0, 0, 0.3)',
    bottom: 0,
    display: 'flex',
    justifyContent: 'center',
    left: 0,
    position: 'fixed',
    right: 0,
    top: 0,
    zIndex: 7,
  },
};

interface MediaModalState {
  isFetching: boolean;
}
class MediaModal extends React.Component<RouteComponentProps, MediaModalState> {
  constructor(props) {
    super(props);
    this.state = {
      isFetching: true,
    };
  }

  render() {
    const { isFetching } = this.state;
    const { history } = this.props;
    return (
      <ReactModal
        isOpen={true}
        onRequestClose={e => {
          history.goBack();
        }}
        style={customStyles}
      >
        <Loader isFetching={isFetching} />
        <ReactPlayer
          width="100%"
          onReady={() => this.setState({ isFetching: false })}
          onError={() => this.setState({ isFetching: false })}
          style={{ display: isFetching ? 'none' : undefined }}
          height="100%"
          url="https://player.vimeo.com/video/223376401?color=00a650&title=0&byline=0&portrait=0"
          config={{
            vimeo: {
              playerOptions: {
                height: 200,
              },
            },
          }}
        />
      </ReactModal>
    );
  }
}

export default MediaModal;
