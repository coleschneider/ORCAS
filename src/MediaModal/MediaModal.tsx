import * as React from 'react';
import ReactModal from 'react-modal';
import ReactPlayer from 'react-player';

(() => {
  if (process.env.NODE_ENV === 'test') {
    return;
  }
  ReactModal.setAppElement('#root');
})();

const customStyles = {
  content: {
    borderRadius: '5px',
    flexShrink: 1,
    overflowY: 'hidden',
    overflowX: 'hidden',
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'stretch',
    justifyContent: 'center',
    top: '50%',
    left: '50%',
    // width: '1000px',
    right: 'auto',
    bottom: 'auto',
    height: '80vh',
    transform: 'translate(-50%, -50%)',
  },
  overlay: {
    zIndex: 7,
    left: 0,
    bottom: 0,
    right: 0,
    top: 0,
    justifyContent: 'center',
    position: 'fixed',
    display: 'flex',
    backgroundColor: 'rgba(0, 0, 0, 0.3)',
  },
};

const MediaModal = ({ history }) => {
  return (
    <ReactModal
      isOpen={true}
      onRequestClose={e => {
        // e.preventDefault();
        // e.stopPropagation();
        history.goBack();
      }}
      style={customStyles}
    >
      <ReactPlayer
        width="100%"
        height="100%"
        url="https://player.vimeo.com/video/223376401?color=00a650&title=0&byline=0&portrait=0"
        config={{
          vimeo: {
            playerOptions: {
              height: 200,
            },
          },
        }}
      />
    </ReactModal>
  );
};

export default MediaModal;
