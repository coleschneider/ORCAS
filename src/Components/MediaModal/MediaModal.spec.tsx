import * as React from 'react';
import MediaModal from './MediaModal';
import ReactPlayer from 'react-player';
const MediaComponent = () => (
  <div>
    Test
    <img />
  </div>
);
describe('MediaModal', () => {
  let historySpy = jest.fn();
  let wrapper;
  const setupModal = (props = { history: historySpy }) => {
    wrapper = shallow(
      <MediaModal {...props}>
        <MediaComponent />
      </MediaModal>,
    );
  };
  it('matches the snapshot', () => {
    setupModal();
    expect(wrapper).toMatchSnapshot();
  });
  it('The Modal renders with a default open prop of true', () => {
    setupModal();
    expect(wrapper.find('Modal').props().isOpen).toBe(true);
  });
  it('The child renders inside of the modal', () => {
    setupModal();
    expect(wrapper.find(MediaComponent).exists()).toBe(true);
  });
});
