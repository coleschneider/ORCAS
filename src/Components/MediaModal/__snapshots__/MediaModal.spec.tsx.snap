// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaModal matches the snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": "5px",
        "bottom": "auto",
        "display": "flex",
        "flexDirection": "column",
        "height": "80vh",
        "justifyContent": "center",
        "left": "50%",
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "absolute",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
      "overlay": Object {
        "backgroundColor": "rgba(0, 0, 0, 0.3)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "center",
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 7,
      },
    }
  }
>
  <MediaComponent />
</Modal>
`;
